internal/cluster/cluster.go

```diff
@@ -437,6 +438,26 @@ func (m *Manager) CancelInstallation(ctx context.Context, c *common.Cluster, rea
 	return nil
 }
 
+func (m *Manager) UpdateInstallProgress(ctx context.Context, c *common.Cluster, progress string) *common.ApiErrorResponse {
```

Comment:
You should probably add unit tests to test bad flows, for example trying to update progress to cluster in error state.

Reply:
Done